### DISCUSSION TITLE
Amazon.Scheduler.Model.ValidationException: 'batch is not a supported service for a target.'

### DISCUSSION THREAD

**Discussion by venukalam** [aws/aws-sdk-net#3134] [Q&A] [answered]
### Describe the bug
I am trying to submit a batch job based on a schedule, using Event bridge scheduler (not using Eventbridge rules because of the max limit of 300 it has)

But when I call the CreateScheduleAsync method with the batch config using .Net SDK, I get a error message saying batch is not supported as a target.

### Expected Behavior
Targetting a batch job through scheduler is possible on the AWS console, So I am expecting it to work with the SDK as well.

### Current Behavior
This is the error message I get - "Amazon.Scheduler.Model.ValidationException: 'batch is not a supported service for a target.'"

### Reproduction Steps
```c#
            var schedulerClient = new AmazonSchedulerClient();
            var scheduleJobInput = new
            {
                JobDefinition = <jobDefinitionName>,
                JobName = <jobName>,
                JobQueue = <jobQueueName>,
                ContainerOverrides = new ContainerOverridesCustom
                {
                    Command = new List<string>() { "some test data" }
                },
                Timeout = new JobTimeoutCustom
                {
                    AttemptDurationSeconds = <timeout>
                }
            };

            var scheduleTimeWindow = new FlexibleTimeWindow
            {
                Mode = FlexibleTimeWindowMode.OFF
            };

            var invokeBatchTarget = new Amazon.Scheduler.Model.Target
            {
                Arn = <jobQueueName>,
                RoleArn = <roleArn>,
                Input = JsonConvert.SerializeObject(scheduleJobInput)
            };

            var createScheduleRequest = new CreateScheduleRequest
            {
                Name = <scheduler-name>,
                ScheduleExpression = "cron(0 12 * * ? *)",
                State = ScheduleState.ENABLED,
                Description = <scheduler-description>,
                FlexibleTimeWindow = scheduleTimeWindow,
                Target = invokeBatchTarget,
                GroupName = <scheduler-group-name>               
            };
            var createScheduleResponse = await schedulerClient.CreateScheduleAsync(createScheduleRequest); 
```

### AWS .NET SDK and/or Package version used
AWSSDK.Scheduler- 3.7.300.16

### Targeted .NET Platform
.NET 6.0

### Operating System and version
Windows 10

**Comment by dscpinheiro** [comment]
Are you able to create the schedule using the equivalent [AWS CLI command](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/scheduler/create-schedule.html)? 

Our client is generated from the service model, we don't explicitly throw that validation exception.

**Reply by venukalam** [reply]
I get the same issue from the equivalent [AWS CLI command](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/scheduler/create-schedule.html)

"An error occurred (ValidationException) when calling the CreateSchedule operation: batch is not a supported service for a target."
 
![Screenshot 2023-12-21 180352](https://github.com/aws/aws-sdk-net/assets/70128175/aa166b58-d313-4fa9-8b03-621ee569f9d4)

**Reply by dscpinheiro** [reply, marked as answer]
Thanks for taking a look. 

What did you specify for the `Amazon.Scheduler.Model.Target.Arn` property? I checked and the console uses `arn:aws:scheduler:::aws-sdk:batch:submitJob`.

If I use this snippet the scheduler is created successfully:
```
var invokeBatchTarget = new Amazon.Scheduler.Model.Target
{
    Arn = "arn:aws:scheduler:::aws-sdk:batch:submitJob",
    RoleArn = "arn:aws:iam::123456789012:role/test-batch-scheduler-role",
    Input = JsonConvert.SerializeObject(scheduleJobInput)
};

// Existing code

var createScheduleResponse = await schedulerClient.CreateScheduleAsync(createScheduleRequest);
Console.WriteLine(createScheduleResponse.HttpStatusCode); // Outputs OK
```

**Reply by venukalam** [reply]
Thanks @dscpinheiro .
Setting Arn to "arn:aws:scheduler:::aws-sdk:batch:submitJob" works.

I was referring to these [examples](https://docs.aws.amazon.com/scheduler/latest/UserGuide/schedule-types.html#:~:text=hours%20%7C%20days-,Examples,-The%20following%20example) and thought we needed to use the batch queue ARN. I also tried using the job definition as the ARN, but that didn't work.

 I didn't realize this is the way.

**Reply by dscpinheiro** [reply]
No problem, it wasn't intuitive at first for me either. 

I left feedback on that documentation page, asking the service team to add more examples where the target is a service operation.
